src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -32,12 +32,11 @@
 
 import static com.github.kagkarlsson.scheduler.ExecutorUtils.defaultThreadFactoryWithPrefix;
 
-public class Scheduler implements SchedulerClient {
+public class Scheduler extends SchedulerClient.StandardSchedulerClient {
 
```

Comment:
Because the implementation for was the same for both of them (`Scheduler` and `StandardSchedulerClient`).

Given the fact that `StandardSchedulerClient` implements `SchedulerClient` you will still be able to "downgrade" the full Scheduler object to only a SchedulerClient.

My reasoning was that if `Scheduler` extends `StandardSchedulerClient`:
* they will share the implementation so we don't repeat ourselves
* No new functionality will be exposed on `Scheduler` as `StandardSchedulerClient` only implements `SchedulerClient`

WDYT?